### ISSUE TITLE
你好，关于mini llama2的几个问题

### ISSUE THREAD

**Issue by lly277365** [chenyangMl/llama2.c-zh#3]
llama2.c-zh是非常棒的项目，小白我有几个问题请教一下：
1、未执行 `python tinystories.py download`和`python tinystories.py pretokenize`时，执行`python train.py`会导致死循环，经检查可能是
tinystories.py的PretokDataset方法里shard_filenames为空列表，导致死循环
2、在win11下运行`python tinystories.py download`时会报错，经检查可能打开文件要加 `encoding='UTF-8'` 编码，在linux下就正常
3、在train.py中，`loss = loss / gradient_accumulation_steps` ，但在 nanoGPT 中，并不需要除以 gradient_accumulation_steps，请问是什么原因呢？
再次感谢llama2.c-zh项目

**Comment by chenyangMl** [comment]
> 

感谢您的反馈,问题1,2代码已修复.

问题3, 这个等式是和[nanoGPT, 第298行](https://github.com/karpathy/nanoGPT/blob/master/train.py)保持一致的.

从逻辑上也能理解. 假设你的batchsize=2, gradient_accumulation_steps=4. 等价于你累加4次batchszie=2的梯度在统一做一次梯度更新.

看下train.py里面第107行, 每次输入的tokens(假设你单卡, ddp_world_size=1).
tokens_per_iter =  梯度累加次数(4) * 1 * bathsize(2) * max_seq_len(256) 
tokens_per_iter = gradient_accumulation_steps * ddp_world_size * batch_size * max_seq_len
